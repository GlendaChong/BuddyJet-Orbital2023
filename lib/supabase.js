import 'react-native-url-polyfill/auto'
import { createClient } from "@supabase/supabase-js";
import AsyncStorage from '@react-native-async-storage/async-storage'

import * as SecureStore from 'expo-secure-store'

const ExpoSecureStoreAdapter = {
      getItem: (key) => {
        return SecureStore.getItemAsync(key)
      },
      setItem: (key, value) => {
        SecureStore.setItemAsync(key, value)
      },
      removeItem: (key) => {
        SecureStore.deleteItemAsync(key)
      },
    }
    

const projectUrl = process.env.SUPABASE_PROJECT_URL
const projectKey = process.env.SUPABASE_PROJECT_KEY

export const supabase = createClient(projectUrl, projectKey, {
    auth: {
        // localStorage: AsyncStorage,
        storage: ExpoSecureStoreAdapter, 
        autoRefreshToken: true,
        persistSession: true,
        detectSessionInUrl: false,
    }
});