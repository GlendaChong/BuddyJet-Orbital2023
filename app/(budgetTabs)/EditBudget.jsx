import { Text } from "react-native"; 
import { SafeAreaView } from "react-native-safe-area-context";
import TextFieldInput from "./TextFieldInput";
import BackButton from "../components/BackButton";
import { ScrollView } from "react-native-gesture-handler";
import { supabase } from "../../lib/supabase";

function Dashboard() {
    return (
        <SafeAreaView style={{ flex: 1, justifyContent: "center" }}>
            <ScrollView>
                <BackButton />
                <Text style={{fontFamily:'Poppins-SemiBold', fontSize:35, marginTop: 40, marginHorizontal: 30}}>Edit</Text>
            </ScrollView>
        </SafeAreaView>
    ); 
}

export default Dashboard; 