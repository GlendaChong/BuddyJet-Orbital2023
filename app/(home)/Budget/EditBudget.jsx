import { Text, View, TextInput, StyleSheet, Button } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";
import BackButton from "../../components/BackButton";
import { ScrollView } from "react-native-gesture-handler";
import { supabase } from "../../../lib/supabase";
import { useState, useEffect } from 'react';
import { GetCategoryDetails, GetCurrentBudget, GetSideHustles, GetUserId } from "../GetBackendData";
import { useLocalSearchParams } from "expo-router";

function EditBudget() {

  const [budgetId, setBudgetId] = useState(0); 
  const [userId, setUserId] = useState(''); 
  const [category, setCategory] = useState([]);
  const [editing, setEditing] = useState(false);
  const [oldIncome, setOldIncome] = useState(0);
  const [newIncome, setNewIncome] = useState(0);
  const [sideHustles, setSideHustles] = useState([]);
  const { selectedMonth, selectedYear } = useLocalSearchParams(); 

  // Fetch backend data 
  const fetchBudgetData = async () => {
    const budget = await GetCurrentBudget(selectedMonth, selectedYear);
    setUserId(budget.user_id); 
    setBudgetId(budget.budget_id); 
    setOldIncome(budget.income);
  }

  const fetchCategoryDetail = async () => {
    const categories = await GetCategoryDetails(selectedMonth, selectedYear);
    setCategory(categories); 
  };

  // Fetch the side hustles from Supabase
  const fetchSideHustles = async () => {
    const sideHustles = await GetSideHustles(selectedMonth, selectedYear); 
    setSideHustles(sideHustles); 
  };

  useEffect(() => {
    fetchBudgetData(); 
  }, []);

  useEffect(() => {
    fetchCategoryDetail();
  }, [budgetId]); 

  useEffect(() => {
    fetchSideHustles();
  }, []);


  // Update income in backend
  const updateIncome = async () => {
    try {
      await supabase
        .from('budget')
        .update({ income: newIncome, in_use: true })
        .eq('budget_id', budgetId)
        .eq('user_id', userId)
        .gte('created_at', `${selectedYear}-${selectedMonth}-01`)
        .lte('created_at', `${selectedYear}-${selectedMonth}-31`);
      // console.log(newIncome); 
      console.log('Income updated successfully');
    } catch (error) {
      console.error('Error updating income:', error.message);
    }
  };

  // Income component 
  const Income = () => {
    const handleEdit = () => {
      setEditing(true);
    };

    const handleSave = () => {
      setOldIncome(newIncome); 
      updateIncome();
      setEditing(false);
    };

    if (editing) {
      return (
        <View style={styles.container}>
          <View style={styles.incomeContainer}>
            <Text style={styles.labelText}>New Income:</Text>
            <TextInput
              value={newIncome.toString()}
              onChangeText={setNewIncome}
              keyboardType="numeric"
              style={styles.input}
              autoFocus
            />
            <Text style={{ marginTop: -30, left: 70 }} onPress={handleSave}>Save</Text>
          </View>
        </View>
      );
    }

    return (
      <View style={{ alignItems: 'center', marginTop: 30 }}>
        <View style={{ backgroundColor: "#F3F6FA", borderRadius: 20, width: 280, height: 90, justifyContent: 'center', alignItems: 'center', marginBottom: 0 }}>
          <Text style={{ color: '#2C2646', fontFamily: 'Poppins-Medium', fontWeight: '600', fontSize: 18, lineHeight: 20, textAlign: 'center' }}>Fixed Income:</Text>
          <Text style={{ color: '#2C2646', fontFamily: 'Poppins-SemiBold', fontWeight: '600', fontSize: 24, lineHeight: 26, textAlign: 'center', marginTop: 15 }}>${oldIncome}</Text>
        </View>
        <Text onPress={handleEdit} style={{ bottom: 38, left: 100 }}>Edit</Text>
      </View>
    );
  };


  // Update category in backend
  const updateCategory = async (updatedCategories) => {
    try {
      await Promise.all(
        updatedCategories.map((category) =>
          supabase
            .from('categories')
            .update({ category: category.category, spending: category.spending })
            .eq('budget_id', budgetId)
            .eq('user_id', userId)
            .eq('category_id', category.category_id)
            .eq('in_use', true)
            .gte('created_at', `${selectedYear}-${selectedMonth}-01`)
            .lte('created_at', `${selectedYear}-${selectedMonth}-31`)
        )
      );
      console.log('Categories updated successfully');
      
    } catch (error) {
      console.error('Error updating categories:', error.message);
    }
  };

  
  // Category box component 
  const SpendingBox = () => {
    const [percentages, setPercentages] = useState([]);
    const [errorMessage, setErrorMessage] = useState('');

    // Initialize the percentages array with default values based on the category data
    useEffect(() => {
      const initialPercentages = category.map((item) => ({
        ...item, 
        spending: parseInt(item.spending * 100),
      }));
      setPercentages(initialPercentages);
    }, []);

    // Update the percentage value for a specific category
    const handlePercentageChange = (category, value) => {
      const updatedPercentages = percentages.map((item) =>
        (item.category == category) ? { ...item, spending: value } : item
      );
      setPercentages(updatedPercentages);
    };

    const handleSave = async () => {
      // Calculate the total sum of percentages
      const totalPercentage = percentages.reduce((sum, item) => sum + parseFloat(item.spending || 0), 0);

      // Perform the validation
      if (totalPercentage != 100) {
        setErrorMessage('Percentage sum is not equal to 100.');
        return;
      }

      const updatedCategories = category.map((item, index) => ({
        ...item,
        category: percentages[index]?.category || item.category,
        spending: parseInt(percentages[index]?.spending) / 100 || item.spending,
      }));

      setCategory(updatedCategories); 
      setErrorMessage('');
      updateCategory(updatedCategories);    
      console.log('Updated categories', category);    
    };

    return (
      <View style={{ marginTop: 30, flex: 1, left: 35 }}>
        <Text style={{ fontFamily: 'Poppins-Medium', fontSize: 18, marginBottom: 15 }}> Categories:</Text>
        <View
          style={{
            backgroundColor: '#F3F6FA',
            borderRadius: 18,
            paddingLeft: 30,
            paddingTop: 20,
            paddingBottom: 8,
            width: 320,
          }}
        >
          {category.map((item, index) => (
            <View key={index} style={styles.categoryContainer}>
              <View style={{ width: 10, height: 10, borderRadius: 10, marginRight: 20, backgroundColor: item.color }} />
              <Text style={styles.categoryText}>{item.category}</Text>
              <TextInput
                value={percentages.find((p) => p.category == item.category)?.spending.toString()}
                onChangeText={(value) => handlePercentageChange(item.category, value)}
                keyboardType="numeric"
                style={styles.percentageText}
              />
              <Text style={styles.percentageText}>%</Text>
            </View>
          ))}
        </View>
        <Text
          onPress={handleSave}
          style={styles.saveText}
        >
          Save
        </Text>
        {errorMessage !='' && <Text style={{ color: 'red', width: 250, top: -20 }}>{errorMessage}</Text>}
      </View>
    );
  };


  // Money in component 
  const MoneyIn = () => {
    const [newSideHustle, setNewSideHustle] = useState('');
    const [newSideHustleAmount, setNewSideHustleAmount] = useState('');

    const addSideHustle = async () => {
      try {
        await supabase
          .from('moneyIn')
          .insert([{ 
            created_at: `${selectedYear}-${selectedMonth}-01`, 
            user_id: userId, 
            name: newSideHustle, 
            amount: newSideHustleAmount 
          }]); 
          
      } catch (error) {
        console.log('Error adding side hustle:', error.message);
      } finally {
        fetchSideHustles(); 
        setNewSideHustle(''); 
        setNewSideHustleAmount(''); 
      }
    };

    return (
      <View style={styles.moneyInOuterContainer}>
        <Text style={styles.moneyInText}>Money In:</Text>
        <View style={styles.moneyInInnerContainer}>
          {sideHustles.map((sideHustle) => (
            <View key={sideHustle.name} style={{ flexDirection: 'row', justifyContent: 'space-between', marginBottom: 20 }}>
              <Text style={styles.sideHustleNameText}>{sideHustle.name}</Text>
              <Text style={styles.sideHustleAmountText}>${sideHustle.amount}</Text>
            </View>
          ))}
          <View style={{ flexDirection: 'row', justifyContent: 'space-between' }}>
            <TextInput
              value={newSideHustle}
              onChangeText={setNewSideHustle}
              placeholder="Money in"
              style={styles.sideHustleNameText}
            />
            <TextInput
              value={newSideHustleAmount}
              onChangeText={setNewSideHustleAmount}
              placeholder="Amount"
              style={styles.sideHustleAmountText}
            />
          </View>
          <View style={{ marginTop: 30 }}>
            <View style={{ borderBottomWidth: 0.5, borderBottomColor: '#EAE9F0', width: 312, right: 30, marginBottomBottom: 10 }} />
            <Button title="Add" onPress={addSideHustle} />
          </View>
        </View>
      </View>
    );
  };


  // Main overall page
  return (
    <SafeAreaView style={{ flex: 1, justifyContent: "center", backgroundColor: 'white' }}>
      <ScrollView>
        <BackButton />
        <Text style={{ fontFamily: 'Poppins-SemiBold', fontSize: 35, marginTop: 40, marginHorizontal: 30 }}>Edit</Text>
        <Income />
        <SpendingBox />
        <MoneyIn />
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    marginTop: 30,
  },
  incomeContainer: {
    backgroundColor: '#F3F6FA',
    borderRadius: 20,
    width: 280,
    height: 110,
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 0,
  },
  labelText: {
    color: '#2C2646',
    fontFamily: 'Poppins-Medium',
    fontWeight: '600',
    fontSize: 18,
    lineHeight: 20,
    textAlign: 'center',
    marginTop: -8,
  },
  input: {
    borderWidth: 1,
    borderColor: '#2C2646',
    borderRadius: 5,
    width: 200,
    height: 40,
    marginTop: 10,
    paddingHorizontal: 10,
  },
  editButton: {
    marginTop: 10,
  },
  editButtonText: {
    color: '#3D70FF',
    fontFamily: 'Poppins-SemiBold',
    fontSize: 18,
  }, 
  categoryContainer: { 
    flexDirection: 'row',
    alignItems: 'center', 
    marginBottom: 15, 
    justifyContent: 'space-between',  
  }, 
  categoryColor: { 
    width: 10, 
    height: 10, 
    borderRadius: 5, 
    marginRight: 20 
  }, 
  categoryText: {
    fontFamily: 'Poppins-SemiBold', 
    width: 120, 
    color: '#2C2646', 
    fontSize: 16, 
  }, 
  percentageText: {
    fontFamily: 'Poppins-Medium', 
    width: 30, 
    color: '#2C2646', 
    fontSize: 16, 
  }, 
  saveText: { 
    fontFamily: 'Poppins-Medium', 
    fontSize: 15, 
    color: '#2C2646', 
    marginTop: 10, 
    left: 260 
  }, 

  moneyInText: { 
    fontFamily: 'Poppins-Medium', 
    fontSize: 18, 
    marginBottom: 8, 
    left: 5 
  }, 
  moneyInOuterContainer: { 
    marginTop: 30,
    paddingHorizontal: 40, 
    alignContent: 'center' 
  }, 
  moneyInInnerContainer: { 
    backgroundColor: '#F3F6FA', 
    borderRadius: 18, 
    paddingHorizontal: 30, 
    paddingTop: 20, 
    marginTop: 10
  }, 
  sideHustleNameText: { 
    fontFamily: 'Poppins-Medium',
    fontSize: 17, 
  }, 
  sideHustleAmountText: {
    fontFamily: 'Poppins-SemiBold', 
    fontSize: 18, 
  }, 
});


export default EditBudget; 