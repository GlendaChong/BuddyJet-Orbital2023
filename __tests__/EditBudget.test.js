import React from "react";
import { render, fireEvent, act } from "@testing-library/react-native";
import EditBudget from "../app/(home)/Budget/EditBudget";
import { supabase } from "../lib/supabase";

// Mock the necessary modules
// Mock the sideHustles data for testing
jest.mock("../lib/supabase", () => ({
  supabase: {
    from: jest.fn().mockReturnThis(),
    insert: jest.fn().mockResolvedValue({}),
    select: jest.fn().mockReturnThis(),
    eq: jest.fn().mockReturnThis(),
    gte: jest.fn().mockReturnThis(),
    lt: jest.fn().mockReturnThis(),
    delete: jest.fn().mockResolvedValue({}),
    mockReturnValueOnce: jest.fn(),
    update: jest.fn().mockResolvedValue({}),
  },
}));
jest.mock("expo-router", () => ({
  useRouter: jest.fn().mockReturnValue({
    push: jest.fn(),
    navigate: jest.fn(),
    goBack: jest.fn(),
  }),
  useLocalSearchParams: jest.fn().mockReturnValue({
    selectedMonth: "sample-month",
    selectedYear: "sample-year",
    monthIndex: "sample-month-index",
  }),
}));

// Mock the functions
jest.mock("../app/components/GetBackendData", () => ({
  GetCategoryDetails: jest.fn().mockResolvedValue([
    { category_id: 1, category: "Category 1", spending: 20 },
    { category_id: 2, category: "Category 2", spending: 30 },
    // Add more category data as needed
  ]),
  GetCurrentBudget: jest.fn().mockResolvedValue({
    user_id: "sample-user-id",
    budget_id: 1,
    income: 1500,
    // Add other properties as needed
  }),
  GetSideHustles: jest.fn().mockResolvedValue([
    { id: 1, name: "Side Hustle 1", amount: 100 },
    { id: 2, name: "Side Hustle 2", amount: 200 },
    // Add more side hustle data as needed
  ]),
}));

jest.mock(
  "react-native/Libraries/Components/Touchable/TouchableOpacity",
  () => {
    const { View } = require("react-native");
    const TouchableOpacityMock = (props) => {
      return <View {...props} />;
    };
    TouchableOpacityMock.displayName = "TouchableOpacity";

    return TouchableOpacityMock;
  }
);

describe("EditBudget", () => {
  it("should render and match snapshot", () => {
    const { toJSON } = render(<EditBudget />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("should update income on edit and save", () => {
    const { getByTestId } = render(<EditBudget />);

    act(() => {
      // Find and click the Edit button for Income
      const editButton = getByTestId("EditButton");
      fireEvent.press(editButton);
    });

    act(() => {
      // Find the TextInput for new income and update its value
      const newIncomeInput = getByTestId("IncomeText");
      fireEvent.changeText(newIncomeInput, "2000");
    });

    // Find and click the Save button for Income
    const saveButton = getByTestId("SaveButton");
    fireEvent.press(saveButton);

    // Ensure that the income is updated and the "Income updated successfully" message is logged
    expect(supabase.from().update).toHaveBeenCalledWith({
      income: "2000",
      in_use: true,
    });
  });

  it("should add money in and update the state and backend correctly", () => {
    const { getByTestId } = render(<EditBudget />);

    act(() => {
      // Find the "Money in" TextInput elements and enter test values
      const moneyInNameInput = getByTestId("SideHustleNameInput");
      fireEvent.changeText(moneyInNameInput, "Test Money In");
    });

    act(() => {
      const moneyInAmountInput = getByTestId("SideHustleAmountInput");
      fireEvent.changeText(moneyInAmountInput, "100");
    });

    // Find and click the "Add" button
    const addButton = getByTestId("AddSideHustleButton");
    fireEvent.press(addButton);

    // Ensure that the money in is added to the state and backend correctly
    expect(supabase.from().insert).toHaveBeenCalledWith([
      {
        created_at: expect.any(String),
        user_id: expect.any(String),
        name: "Test Money In",
        amount: "100",
      },
    ]);
  });

  it("should display an error message for an invalid total percentage", () => {
    // Render the component
    const { getByTestId, getByText } = render(<EditBudget />);
    act(() => {
      // Find the save button and simulate a click
      const saveButton = getByTestId("CategoriesSave");
      fireEvent.press(saveButton);
    });

    // Assert that the error message is displayed
    const errorMessage = getByText("Percentage sum is not equal to 100.");
    expect(errorMessage).toBeDefined();
  });

  // You can write additional tests for other functionalities in a similar manner
});
