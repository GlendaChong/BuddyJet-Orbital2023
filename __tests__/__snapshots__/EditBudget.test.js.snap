// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditBudget should render and match snapshot 1`] = `
<RNCSafeAreaView
  edges={
    [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView
      collapsable={false}
      handlerTag={1}
      handlerType="NativeViewGestureHandler"
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      waitFor={
        [
          {
            "current": null,
          },
        ]
      }
    >
      <View>
        <Svg
          focusable={false}
          height={20}
          style={
            {
              "left": 30,
              "marginTop": 7,
              "position": "absolute",
            }
          }
          viewBox="0 0 320 512"
          width={20}
        >
          <Path
            d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l192 192c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256 246.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-192 192z"
            fill="#000"
          />
        </Svg>
        <Text
          onPress={[Function]}
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              {
                "writingDirection": "ltr",
              },
              {
                "color": "#fff",
                "fontFamily": "Poppins-Regular",
                "fontSize": 15,
                "fontWeight": 600,
                "left": 20,
                "marginTop": 7,
                "opacity": 0.1,
                "position": "absolute",
              },
            ]
          }
        >
          back
        </Text>
        <Text
          style={
            {
              "fontFamily": "Poppins-SemiBold",
              "fontSize": 35,
              "marginHorizontal": 30,
              "marginTop": 40,
            }
          }
        >
          Edit
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "marginTop": 30,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#F3F6FA",
                "borderRadius": 20,
                "height": 90,
                "justifyContent": "center",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.4,
                "shadowRadius": 5,
                "width": 280,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#2C2646",
                  "fontFamily": "Poppins-Medium",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "lineHeight": 20,
                  "textAlign": "center",
                }
              }
            >
              Fixed Income:
            </Text>
            <Text
              style={
                {
                  "color": "#2C2646",
                  "fontFamily": "Poppins-SemiBold",
                  "fontSize": 24,
                  "fontWeight": "600",
                  "lineHeight": 26,
                  "marginTop": 15,
                  "textAlign": "center",
                }
              }
            >
              $
              0
            </Text>
          </View>
          <Text
            onPress={[Function]}
            style={
              {
                "bottom": 42,
                "fontFamily": "Poppins-Regular",
                "left": 100,
              }
            }
            testID="EditButton"
          >
            Edit
          </Text>
        </View>
        <View
          style={
            {
              "flex": 1,
              "marginTop": 30,
              "paddingHorizontal": 30,
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Poppins-Medium",
                "fontSize": 18,
                "marginBottom": 15,
              }
            }
          >
             Categories:
          </Text>
          <View
            style={
              {
                "backgroundColor": "#F3F6FA",
                "borderRadius": 18,
                "paddingBottom": 8,
                "paddingTop": 20,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.4,
                "shadowRadius": 5,
              }
            }
          />
          <Text
            onPress={[Function]}
            style={
              {
                "color": "#2C2646",
                "fontFamily": "Poppins-Medium",
                "fontSize": 15,
                "marginTop": 10,
                "right": 10,
                "textAlign": "right",
              }
            }
            testID="CategoriesSave"
          >
            Save
          </Text>
        </View>
        <View
          style={
            {
              "alignContent": "center",
              "marginTop": 30,
              "paddingBottom": 20,
              "paddingHorizontal": 30,
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Poppins-Medium",
                "fontSize": 18,
                "left": 5,
                "marginBottom": 8,
              }
            }
          >
            Money In:
          </Text>
          <View
            style={
              {
                "backgroundColor": "#F3F6FA",
                "borderRadius": 18,
                "marginTop": 10,
                "paddingHorizontal": 30,
                "paddingTop": 20,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.4,
                "shadowRadius": 5,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <TextInput
                onChangeText={[Function]}
                placeholder="Money in"
                style={
                  {
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 17,
                  }
                }
                testID="SideHustleNameInput"
                value=""
              />
              <TextInput
                onChangeText={[Function]}
                placeholder="Amount"
                style={
                  {
                    "fontFamily": "Poppins-SemiBold",
                    "fontSize": 18,
                    "marginLeft": "auto",
                    "right": 30,
                  }
                }
                testID="SideHustleAmountInput"
                value=""
              />
            </View>
            <View
              style={
                {
                  "marginTop": 30,
                }
              }
            >
              <View
                style={
                  {
                    "borderBottomColor": "#8E8E93",
                    "borderBottomWidth": 0.5,
                    "opacity": 0.7,
                    "paddingHorizontal": 5,
                  }
                }
              />
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                onPress={[Function]}
                testID="AddSideHustleButton"
              >
                <View
                  style={
                    [
                      {},
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "margin": 8,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    Add
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaView>
`;
